src/stc/PlatWX.cpp

```diff
@@ -2541,20 +2521,31 @@ void ListBoxImpl::RegisterImageHelper(int type, wxBitmap& bmp)
 }
 
 void ListBoxImpl::RegisterImage(int type, const char *xpm_data) {
-    wxMemoryInputStream stream(xpm_data, strlen(xpm_data)+1);
-    wxImage img(stream, wxBITMAP_TYPE_XPM);
+    wxXPMDecoder dec;
+    wxImage img;
+
+    // This check is borrowed from src/stc/scintilla/src/XPM.cpp.
```

Comment:
As mentioned on the mailing list, could we please replace this with `strncmp(xpm_data, "/* XPM */", 9) == 0`? Or am I missing some reason for which this simpler test wouldn't work?

Reply:
I believe the idea behind that check is that the function can be either passed a set of XPM data, a C array of C strings, or a NULL terminated copy of an XPM file.  The [Scintilla documentation](https://www.scintilla.org/ScintillaDoc.html#Images) refers to these as "lines form" and "text form".  To try and tell which kind of data has been input, on a 32 bit system it's safe to look at the first 4 bytes and on a 64 bit system it's safe to look at the first 8 bytes - but I think it just picks the smaller number since it will work with both.  If those first 4 bytes are not '/' '*' ' ' 'X', then there's data sent could be in "text form" and we shouldn't look any further at the contents of that memory location.  Instead we should that memory contents as a pointer to first string of the XPM data.

I'm not 100% sure but I believe that's why the more complicated check is used in xpm.cpp instead of the simpler one you're suggesting.

Reply:
Do we need to support such polymorphism? Can't we only accept real `const char*` here and not either it or `const char* const*`?

Reply:
This function is what ultimately does the processing for the SendMsg method with the SCI_REGISTERIMAGE message.  Both of these calls should work with wxSTC

```
char * somedata_xpm[] = .... (some xpm data here)
m_stc->SendMsg(SCI_REGISTERIMAGE,0,reinterpret_cast<wxIntPtr>(somedata_xpm));
```

```
const char* someXMPfile = ... (load an xpm an xmp file here)
m_stc->SendMsg(SCI_REGISTERIMAGE,0,reinterpret_cast<wxIntPtr>(someXMPfile));
```
Both of those calls will wind through several different files in both the STC and Scintilla sourced and ultimately end up in this method.

Reply:
OK, let's merge it like this but this is really pretty ugly and I'm almost sure that the original problem (which forced you to update this code) is somehow related to mixing up string pointers and arrays of string pointers somewhere. I'd also feel better if you could compile (and then run) the sample with ASAN and/or under Valgrind...